spec.html

```diff
@@ -1111,30 +1111,857 @@ <h1>Well-Known Symbols</h1>
+            1. Return a BigInt representing _x_ &times; 2<sup>_y_</sup>.
+          </emu-alg>
+          <emu-note>Semantics here should be equivalent to a bitwise shift, treating the BigInt as an infinite length string of binary two's complement digits.</emu-note>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-bigint-signedRightShift">
+          <h1>BigInt::signedRightShift ( _x_, _y_ )</h1>
+          <p>The abstract operation BigInt::signedRightShift with arguments _x_ and _y_ of type BigInt performs the following steps:</p>
+          <emu-alg>
+            1. Return BigInt::leftShift(_x_, -_y_).
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-bigint-unsignedRightShift">
+          <h1>BigInt::unsignedRightShift ( _x_, _y_ )</h1>
+          <p>The abstract operation BigInt::unsignedRightShift with two arguments _x_ and _y_ of type BigInt performs the following steps:</p>
+          <emu-alg>
+            1. Throw a *TypeError* exception.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-bigint-lessThan">
+          <h1>BigInt::lessThan ( _x_, _y_ )</h1>
+          <p>The abstract operation BigInt::lessThan with two arguments _x_ and _y_ of type BigInt returns *true* if _x_ is less than _y_ and *false* otherwise.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-bigint-equal">
+          <h1>BigInt::equal ( _x_, _y_ )</h1>
+          <p>The abstract operation BigInt::equal with two arguments _x_ and _y_ of type BigInt returns *true* if _x_ and _y_ have the same mathematical integer value and *false* otherwise.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-bigint-sameValue">
+          <h1>BigInt::sameValue ( _x_, _y_ )</h1>
+          <p>The abstract operation BigInt::sameValue with two arguments _x_ and _y_ of type BigInt performs the following steps:</p>
+          <emu-alg>
+            1. Return BigInt::equal(_x_, _y_).
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-bigint-sameValueZero">
+          <h1>BigInt::sameValueZero ( _x_, _y_ )</h1>
+          <p>The abstract operation BigInt::sameValueZero with two arguments _x_ and _y_ of type BigInt performs the following steps:</p>
+          <emu-alg>
+            1. Return BigInt::equal(_x_, _y_).
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-bigintbitwiseop" aoid="BigIntBitwiseOp">
+          <h1>BigIntBitwiseOp ( _op_, _x_, _y_ )</h1>
+          <emu-alg>
+            1. Let _result_ be *0n*.
+            1. Let _shift_ be 0.
+            1. Repeat, until (_x_ = 0 or _x_ = -1) and (_y_ = 0 or _y_ = -1),
+              1. Let _xDigit_ be _x_ modulo 2.
+              1. Let _yDigit_ be _y_ modulo 2.
+              1. Let _result_ be _result_ + 2<sup>_shift_</sup> &times; _op_(_xDigit_, _yDigit_).
```

Comment:
i'm a little confused how ``` `&amp;` ```, for example, is an abstract operation. Can you elaborate?

Reply:
Thanks for pointing out this. At the first time I saw it, I was thinking in prefix notation, but I think I have a better idea for this.

Reply:
i see another place in the PR/spec where there's an if/else dispatch - ie, it looks at the string, and invokes the right hardcoded abstract op, rather than trying to be metaprogrammy about it

Reply:
This is fair. I changed to this approach.